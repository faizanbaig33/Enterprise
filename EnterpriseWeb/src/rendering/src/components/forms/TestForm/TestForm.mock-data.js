const defaultData ={
  rendering: {
    uid: "333417aa-6a46-43eb-8b6b-f21264373984",
    componentName: "TestForm",
    dataSource: "{A7930EF8-48F0-441F-8784-C606203DF7FC}",
    placeholders: {
      form: [
        {
          uid: "f73943cb-04ad-45d7-a947-e58011646a59",
          componentName: "Form",
          dataSource: "{F73943CB-04AD-45D7-A947-E58011646A59}",
          fields: {
            children: [
              {
                id: "54dac851-a24e-4a69-9df1-cb3383c5b28e",
                displayName: "Page",
                name: "Page",
                templateId: "c4442680-1100-4626-a469-acc318e25cbf",
                templateName: "Page",
                url: "/Data/Forms/Form/Page",
                fields: {
                  children: [
                    {
                      id: "b5616f85-9d93-448f-91df-c22be462c97a",
                      displayName: "Input",
                      name: "Input",
                      templateId: "f57c26c9-3dab-44d4-8bcb-8262137929ac",
                      templateName: "Input",
                      url: "/Data/Forms/Form/Page/Input",
                      fields: {
                        title: {
                          value: "Input Title"
                        }
                      }
                    },
                    {
                      id: "0e5b4283-f9b5-4d86-8b86-ae9ca9d4345a",
                      displayName: "Paragraph",
                      name: "Paragraph",
                      templateId: "1e258863-055b-4c0d-b775-59541b0a90fd",
                      templateName: "Paragraph",
                      url: "/Data/Forms/Form/Page/Paragraph",
                      fields: {
                        text: {
                          value: "This is text in the text area."
                        }
                      }
                    }
                  ]
                }
              }
            ]
          }
        }
      ]
    }
  },
  fields: {
    componentSpacing: null,
    sectionId: {
      value: ""
    },
    theme: {
      value: ""
    },
    eventType: {
      value: ""
    },
    eventZone: {
      value: ""
    },
    eventName: {
      value: ""
    }
  }
  };

  export default defaultData;
