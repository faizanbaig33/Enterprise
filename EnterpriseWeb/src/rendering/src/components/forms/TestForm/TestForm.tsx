import { ComponentRendering, Placeholder } from '@sitecore-jss/sitecore-jss-react';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { Component } from 'src/helpers/Component';

export type TestFormProps = Feature.EnterpriseWeb.Components.Forms.TestForm & {
  rendering: ComponentRendering;
};

const TestForm = (props: TestFormProps): JSX.Element => {
  return (
    <Component variant="lg" dataComponent="forms/testform" {...props}>
      <div className="col-span-12">
        <h1>Test Form</h1>
        <Placeholder name="form" rendering={props.rendering} />
      </div>
    </Component>
  );
};

export default TestForm;
