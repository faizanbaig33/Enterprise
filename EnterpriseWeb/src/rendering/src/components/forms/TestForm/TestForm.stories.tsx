// Global
import { Story, Meta } from '@storybook/react';
// Local
import TestForm, { TestFormProps } from './TestForm';
import defaultData from './TestForm.mock-data';

export default {
  title: 'Forms/TestForm',
  component: TestForm,
} as Meta;
const Template: Story<TestFormProps> = (props) => <TestForm {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
