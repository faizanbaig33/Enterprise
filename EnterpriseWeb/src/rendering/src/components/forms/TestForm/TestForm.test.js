// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import TestForm from './TestForm';
import defaultData from './TestForm.mock-data';

it('renders correctly', () => {
  const component = snapshot(TestForm, { componentProps: defaultData });
  hasDataComponent(component, 'forms/testform');
});
