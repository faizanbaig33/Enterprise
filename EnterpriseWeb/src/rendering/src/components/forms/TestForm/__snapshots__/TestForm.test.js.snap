// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="forms/testform"
  id="id579052649"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="col-span-12"
      >
        <h1>
          Test Form
        </h1>
        <div
          style="background: darkorange; outline: 5px solid orange; padding: 10px; color: white; max-width: 500px;"
        >
          <h2>
            Form
          </h2>
          <p>
            JSS component is missing React implementation. See the developer console for more information.
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
